### ISSUE TITLE
ARM build segfault

### ISSUE THREAD

**Issue by ViralBShah** [JuliaLang/julia#14466]
Building the current master (commit 4493da9608c817bf74fb006405ca07ec9f85b46d) segfaults on arm.

````
    JULIA usr/lib/julia/inference0.ji
WARNING: unable to determine host cpu name.
essentials.jl
reflection.jl
options.jl
promotion.jl
tuple.jl
range.jl
expr.jl
error.jl
bool.jl
number.jl
int.jl
operators.jl
pointer.jl
abstractarray.jl
array.jl
hashing.jl
nofloat_hashing.jl
functors.jl
reduce.jl
intset.jl
dict.jl
iterator.jl
inference.jl
    JULIA usr/lib/julia/inference.ji
WARNING: unable to determine host cpu name.
essentials.jl
reflection.jl
options.jl
promotion.jl
tuple.jl
range.jl

signal (11): Segmentation fault
while loading range.jl, in expression starting on line 718
Segmentation fault
make[1]: *** [/home/viral/julia/usr/lib/julia/inference.ji] Error 139
make: *** [julia-inference] Error 2
````

**Comment by yuyichao** [comment]
Hmm, looking at the backtrace, it looks like `is_assigned_inner` is being miscompiled (by our codegen or llvm). The LLVM IR to get the `vinfo` field of a `StaticVarInfo` is

```llvm
  %12 = getelementptr inbounds %jl_value_t, %jl_value_t* %0, i64 2, i32 0, !dbg !117
  %13 = load %jl_value_t*, %jl_value_t** %12, align 8, !dbg !117, !tbaa !99
```

and I believe the offset in the `gep` should be 4 (5th field) instead of 2.

Trying to see what the IR looks like before the LLVM passes.

**Comment by Keno** [comment]
Hah, I know what the problem is! Should be fixed on my kf/llvmpatches branch.

**Comment by yuyichao** [comment]
Is it fixed on llvm-svn as well?

I'm glade I asked before going further =)

**Comment by Keno** [comment]
It's a julia bug. Missing data layout on some modules, which makes the optimization pass think pointers are of the wrong size.

**Comment by yuyichao** [comment]
So I can still use the system LLVM. That's a good news since I don't have to wait for the LLVM compilation =). I'll try that now.

**Comment by yuyichao** [comment]
Yep, that fixes the problem. Still waiting for the compilation to finish but it passes the point where it previously segfault now =)

**Comment by yuyichao** [comment]
I add this issue to the ones closed by #14430
